src/components/BaseComponents/FormGroup/FieldSet.tsx

```diff
@@ -3,19 +3,25 @@ import React, {useState, useEffect, useContext } from 'react';
 import PropTypes from 'prop-types';
 import ConditionalWrapper from '../../helpers/formatters/ConditionalWrapper';
 import HintTextComponent from '../../helpers/formatters/ParsedHtml';
-import { DefaultFormContext } from '../../helpers/HMRCAppContext';
+import { DefaultFormContext, ErrorMsgContext } from '../../helpers/HMRCAppContext';
 import InstructionComp from '../../helpers/formatters/ParsedHtml';
+import { checkErrorMsgs } from '../../helpers/utils';
 
 export default function FieldSet({legendIsHeading=false, label, name, errorText, hintText, children, fieldsetElementProps, testProps}){
-  const[ErrorMessage,setErrorMessage] = useState(errorText);
+ 
 
   const {instructionText} = useContext(DefaultFormContext);
   
+  const {errorMsgs} = useContext(ErrorMsgContext);
+  const [errMessage,setErrorMessage] = useState(errorText);
 
   useEffect(()=>{
-    setErrorMessage(errorText)
-  },[errorText])
-  const formGroupDivClasses = `govuk-form-group ${ErrorMessage?'govuk-form-group--error':""}`.trim();
+    const found = checkErrorMsgs(errorMsgs,name)
```

Comment:
your 2nd parameter is id but you are passing name here.

Reply:
Yeah on certain Components we are fetching name instead of id . so that is being passed as second param here. In terms of validation also its working fine.